File: src/com/kraftwerking/datastructuresalgo/leetcode/JumpGame55.java
package com.kraftwerking.datastructuresalgo.leetcode;


public class JumpGame55 {

  public boolean canJump(int[] nums) {
    int goal = nums.length - 1;

    for (int i = nums.length - 1; i >= 0; i--) {
      if ((i + nums[i]) >= goal) {
        goal = i;
      }
    }

    if (goal == 0) {
      return true;
    } else {
      return false;
    }
  }

  public static void main(String[] args) {
    int[] nums = {3, 2, 1, 0, 4};

    JumpGame55 jumpGame55 = new JumpGame55();
    boolean result = jumpGame55.canJump(nums);
    System.out.println(result);

  }
}

File: python/JumpGame55.py
from typing import List


class Solution(object):
    def canJump(self, nums: List[int]) -> bool:
        goal = len(nums) - 1

        for i in range(len(nums) - 1, -1, -1):
            if i + nums[i] >= goal:
                goal = i

        return True if goal == 0 else False


def main():
    nums = [2, 3, 1, 1, 4]

    solution = Solution()
    result = solution.canJump(nums)
    print(result)


if __name__ == "__main__":
    main()
